frontend/src/components/ViewActivities/editActivity.js

```diff
@@ -38,27 +38,40 @@ class EditActivity extends React.Component {
     this.editEndTzRef = React.createRef();
   }
   
- /**
+  /**
    * Edit an activity in the database upon form submission.
-   * TODO: Update times as well! This only does the text field forms (#64).
    */
   editActivity() {
+    const activity = this.props.activity;
+
     let newVals = {};
-    if (this.editTitleRef.current.value !== '') {
-      newVals[DB.ACTIVITIES_TITLE] = this.editTitleRef.current.value;
-    }
-    if (this.editDescriptionRef.current.value !== '') {
-      newVals[DB.ACTIVITIES_DESCRIPTION] = this.editDescriptionRef.current.value;
-    }
-    if (this.editStartLocRef.current.value !== 'No Change'){
-      newVals[DB.ACTIVITIES_START_COUNTRY] = this.editStartLocRef.current.value;
-    }
-    if (this.editEndLocRef.current.value !== 'No Change'){
-      newVals[DB.ACTIVITIES_END_COUNTRY] = this.editEndLocRef.current.value;
-    }
-    if (Object.keys(newVals).length !== 0) {
-      writeActivity(this.props.activity.tripId, this.props.activity.id, newVals);
-    }
+    // All the text fields. 
+    newVals[DB.ACTIVITIES_TITLE] = 
+      getRefValue(this.editTitleRef, '', activity[DB.ACTIVITIES_TITLE])
+    newVals[DB.ACTIVITIES_DESCRIPTION] = 
+      getRefValue(this.editDescriptionRef, '', activity[DB.ACTIVITIES_DESCRIPTION]);
+
+    newVals[DB.ACTIVITIES_START_COUNTRY] = 
+      getRefValue(this.editStartLocRef, 'No Change', activity[DB.ACTIVITIES_START_COUNTRY]);
+    newVals[DB.ACTIVITIES_END_COUNTRY] = 
+      getRefValue(this.editEndLocRef, 'No Change', activity[DB.ACTIVITIES_END_COUNTRY]);
+    
+    newVals[DB.ACTIVITIES_START_TZ] = getRefValue(this.editStartTzRef, '', '');
+    newVals[DB.ACTIVITIES_END_TZ] = getRefValue(this.editEndTzRef, '', '');
```

Comment:
So is this saying if the current value of the timezones is an empty string, then you want to return an empty string?  If so and you want to make this less verbose/more clear, you could make the `ignoreValue` parameter have a default value of an empty string and make it the last parameter (see below) to avoid having to specify it in most of your cases

`getRefValue(ref, defaultValue=null, ignoreValue='') `

Reply:
That's a good point.